js/app/modules/contentGroup/mediaLightbox.js

```diff
@@ -66,44 +73,80 @@ const MediaLightbox = new Module.Class({
 
     _lightbox_shift_image: function (delta) {
-        if (this._article_model === null)
+        if (this._context === null)
             return;
         if (this._loading_new_lightbox)
             return;
 
         this._loading_new_lightbox = true;
         let new_index = this._current_index + delta;
-        let resource_id = this._article_model.resources[new_index];
-        Engine.get_default().get_object_by_id(resource_id, null, (engine, task) => {
+        let resource = this._context[new_index];
+        if (resource instanceof ContentObjectModel.ContentObjectModel) {
+            this._preview_media_object(resource);
             this._loading_new_lightbox = false;
-            let media_object;
-            try {
-                media_object = engine.get_object_by_id_finish(task);
-            } catch (error) {
-                logError(error);
-                return;
-            }
+        } else {
+            Engine.get_default().get_object_by_id(resource, null, (engine, task) => {
+                this._loading_new_lightbox = false;
+                let media_object;
+                try {
+                    media_object = engine.get_object_by_id_finish(task);
+                } catch (error) {
+                    logError(error);
+                    return;
+                }
 
-            // If the next object is not the last, the forward arrow should be displayed.
-            this._preview_media_object(media_object);
-            this._loading_new_lightbox = false;
-        });
+                // If the next object is not the last, the forward arrow should be displayed.
+                this._preview_media_object(media_object);
+                this._loading_new_lightbox = false;
+            });
+        }
     },
 
     _preview_media_object: function (media_object) {
-        if (this._article_model === null)
+        if (this._context === null)
             return;
-        let resources = this._article_model.resources;
-        this._current_index = resources.indexOf(media_object.ekn_id);
+        let resources = this._context;
+
+        this._current_index = resources.map((item) => {
+            if (item instanceof ContentObjectModel.ContentObjectModel) {
+                return item.ekn_id;
+            }
+            return item;
+        }).indexOf(media_object.ekn_id);
+
         if (this._current_index === -1)
             return;
 
         if (this.lightbox_widget)
             this.drop_submodule(this.lightbox_widget);
-        let card = this.create_submodule('card', {
+
+        let widget = this.create_submodule('card', {
             model: media_object
         });
-        this.lightbox_widget = card;
+
+        // Lightboxes may need to show a variety of different content, in which
+        // case a single card type may not fit all needs. If no card type is
+        // specified, try to determine content type and show it accordingly.
+        let content_type = media_object.content_type;
```

Comment:
The designs call for this content to be displayed in a lightbox. previewer.js has none of the lightboxing code, so we would have to repeat it all there, not to mention having to create a separate module wrapper class, since previewer is just a widget.